### PR TITLE
fix(react): Fix react wrappers resizing with animation

### LINKED ISSUE #111: Animated initial render in frameworks (e.g. React, angular etc.)
Suspect this is an issue with `requestAnimationFrame`, which should be replaced with a more efficient method such as `ResizeObserver`

### PR DESCRIPTION
**Pull request by theiliad**
Closes #111 

### Updates
- Add width & height props to **react** `BaseChart`
- Support width & height option values for `core` package

### CODE REVIEW

File: packages/core/src/base-chart.ts [hunk 1]
```diff
@@ -51,19 +51,20 @@ export class BaseChart {
...
 		this.container = container;
 		this.chartContainerID = chartId;
 
-		if (configs.options) {
-			this.options = Object.assign({}, this.options, configs.options);
-
-			if (this.options.containerResizable) {
-				this.resizeWhenContainerChange();
-			}
+		if (this.options && this.options.containerResizable) {
```
Comment (zvonimirfras) [inline packages/core/src/base-chart.ts]: Based on what I see here `this.options` should always exist and checking for it is redundant.

File: packages/core/src/base-chart.ts [hunk 2]
```diff
@@ -73,6 +74,33 @@ export class BaseChart {
 		}
 	}
 
+	styleHolderElement() {
+		const holderElement = this.holder as HTMLElement;
+		const { width, height } = this.options;
+
+		// If width exists in options
+		if (width) {
+			// Validate it's value
```
Comment (zvonimirfras) [inline packages/core/src/base-chart.ts]: Should this be
```suggestion
			// Validate its value
```

File: packages/core/src/base-chart.ts [hunk 3]
```diff
@@ -73,6 +74,33 @@ export class BaseChart {
...
+				// Apply formatted width attribute to chart
+				holderElement.style.width = Tools.formatWidthHeightValues(width);
+			} else {
+				console.error("`width` is badly formatted");
+			}
+		}
+
+		// If height exists in options
+		if (height) {
+			// Validate it's value
```
Comment (zvonimirfras) [inline packages/core/src/base-chart.ts]: Should this be
```suggestion
			// Validate its value
```

File: packages/core/src/base-chart.ts [hunk 4]
```diff
@@ -73,6 +74,33 @@ export class BaseChart {
...
+	styleHolderElement() {
+		const holderElement = this.holder as HTMLElement;
+		const { width, height } = this.options;
+
+		// If width exists in options
+		if (width) {
+			// Validate it's value
+			if (Tools.validateWidthHeightValues(width)) {
+				// Apply formatted width attribute to chart
+				holderElement.style.width = Tools.formatWidthHeightValues(width);
```
Comment (zvonimirfras) [inline packages/core/src/base-chart.ts]: I see you're trying to enforce using sizes in % or px but I'm having a hard time understanding why are those only viable options.

File: packages/react/src/base-chart.js [hunk 5]
```diff
@@ -7,6 +7,20 @@ export default class BaseChart extends React.Component {
 		this.data = props.data;
 		this.options = props.options;
 
+		// Width prop is mandatory for the wrappers
```
Comment (zvonimirfras) [inline packages/react/src/base-chart.js]: Why are width and height "mandatory for the wrappers"?

File: packages/core/src/base-chart.ts (hunk 1 above)
Comment (theiliad) [reply packages/core/src/base-chart.ts]: Good catch 👍

File: packages/react/src/base-chart.js (hunk 5 above)
Comment (theiliad) [reply packages/react/src/base-chart.js]: This is an issue with frameworks such as React & Angular. The chart holder does not start at its ideal size initially, and re-renders multiple times to get to the final chart size, an as a result of that, `resizeChart()` is called multiple times, making the initial render seem animated.
image

We can use `react-sizeme` solves this by initially rendering the component invisibly, then re-rendering to the full size, or another approach that **most** charting frameworks use in React is to make width & height mandatory.

**Pull request merged**